tests/unit/test_gce_authority.py

```diff
@@ -0,0 +1,188 @@
+        'crm_scopes': ['crm-scope'],
+        'gce_keyfile': 'gce-keyfile',
+        'gce_scopes': ['gce-scope'],
+        'metadata_blacklist': [['key', 'val'], ['other_key', 'other_val']],
+        'project_blacklist': [],
+        'tag_blacklist': [],
+        'zone': 'zone1',
+    }
+
+
+@pytest.fixture
+def authority(mocker, monkeypatch, authority_config, auth_client):
+    monkeypatch.setattr(
+        'gordon_janitor_gcp.gce_authority.auth.GoogleAuthClient',
+        mocker.Mock(return_value=auth_client))
+    rrset_channel = asyncio.Queue()
+    authority = gce_authority.GCEAuthority(
+        authority_config, rrset_channel)
+    authority.session = auth_client._session
+    return authority
+
+
+@pytest.fixture
+def get_mock_client(mocker, get_gce_client):
+    def _create_fake(klass, *args, **kwargs):
+        client = get_gce_client(klass, *args, **kwargs)
+        fake_client = mocker.Mock(client)
+        return fake_client
+    return _create_fake
+
+
+@pytest.mark.asyncio
+async def test_builder_creates_proper_authority(mocker, authority_config):
+    gce_client = mocker.MagicMock(name='gce_client')
+    crm_client = mocker.MagicMock(name='crm_client')
+    auth_client = mocker.MagicMock(name='auth_client')
+    authority_mod_patch = 'gordon_janitor_gcp.gce_authority.'
+    mocker.patch(authority_mod_patch + 'gcp_clients.GCRMClient', crm_client)
+    mocker.patch(authority_mod_patch + 'gcp_clients.GCEClient', gce_client)
+    mocker.patch(authority_mod_patch + 'auth.GoogleAuthClient', auth_client)
+
+    rrset_channel = 'channel'
+    kwargs = {'other': 'kwargs'}
+    builder = gce_authority.GCEAuthorityBuilder(
+        authority_config, rrset_channel, **kwargs)
+    session = builder.session
+    with contextlib.closing(session):
+        authority = builder.build_authority()
+
+    auth_client_calls = [
+        mocker.call(
+            authority_config['crm_keyfile'],
+            scopes=authority_config['crm_scopes'],
+            session=session),
+        mocker.call(
+            authority_config['gce_keyfile'],
+            scopes=authority_config['gce_scopes'],
+            session=session),
+    ]
+
+    auth_client.assert_has_calls(auth_client_calls, any_order=True)
+    expected_metadata_blacklist = [
+        dict([pair]) for pair in authority_config['metadata_blacklist']
+    ]
+    gce_client.assert_called_once_with(
+        auth_client(),
+        session,
+        tag_blacklist=authority_config['tag_blacklist'],
+        metadata_blacklist=expected_metadata_blacklist)
+
+    assert crm_client() is authority.crm_client
+    assert gce_client() is authority.gce_client
+    assert rrset_channel is authority.rrset_channel
+    assert authority_config is authority.config
+
+
+@pytest.mark.asyncio
+async def test_builder_error_on_required_config(authority_config, caplog):
+    caplog.set_level(logging.ERROR)
+    del authority_config['crm_keyfile']
+    rrset_channel = 'channel'
```

Comment:
Is there a reason why you create a proper queue elsewhere, but just a string here?